src/test/rpc/LedgerEntry_test.cpp

```diff
@@ -31,40 +31,435 @@
+                }
+            }
+            return values;
+        };
+
+        static auto const& badUInt32Values = remove({2, 3});
+        static auto const& badUInt64Values = remove({2, 3});
+        static auto const& badHashValues = remove({2, 3, 7, 8, 16});
+        static auto const& badAccountValues = remove({12});
+        static auto const& badBlobValues = remove({3, 7, 8, 16});
+        static auto const& badCurrencyValues = remove({14});
+        static auto const& badArrayValues = remove({17, 20});
+        static auto const& badIndexValues = remove({12, 16, 18, 19});
+
+        switch (fieldType)
+        {
+            case UInt32Field:
+                return badUInt32Values;
+            case UInt64Field:
+                return badUInt64Values;
+            case HashField:
+                return badHashValues;
+            case AccountField:
+                return badAccountValues;
+            case BlobField:
+                return badBlobValues;
+            case CurrencyField:
+                return badCurrencyValues;
+            case ArrayField:
+            case TwoAccountArrayField:
+                return badArrayValues;
+            case HashOrObjectField:
+                return badIndexValues;
+            default:
+                Throw<std::runtime_error>(
+                    "unknown type " + std::to_string(fieldType));
+        }
+    }
+
+    Json::Value
+    getCorrectValue(Json::StaticString fieldName)
+    {
+        static Json::Value const twoAccountArray = []() {
+            Json::Value arr(Json::arrayValue);
+            arr[0u] = "rhigTLJJyXXSRUyRCQtqi1NoAZZzZnS4KU";
+            arr[1u] = "r4MrUGTdB57duTnRs6KbsRGQXgkseGb1b5";
+            return arr;
+        }();
+
+        auto const typeID = getFieldType(fieldName);
+        switch (typeID)
+        {
+            case UInt32Field:
+                return 1;
+            case UInt64Field:
+                return 1;
+            case HashField:
+                return "5233D68B4D44388F98559DE42903767803EFA7C1F8D01413FC16EE6"
+                       "B01403D6D";
+            case AccountField:
+                return "r4MrUGTdB57duTnRs6KbsRGQXgkseGb1b5";
+            case BlobField:
+                return "ABCDEF";
+            case CurrencyField:
+                return "USD";
+            case ArrayField:
+                return Json::arrayValue;
+            case HashOrObjectField:
+                return "5233D68B4D44388F98559DE42903767803EFA7C1F8D01413FC16EE6"
+                       "B01403D6D";
+            case TwoAccountArrayField:
+                return twoAccountArray;
+            default:
+                Throw<std::runtime_error>(
+                    "unknown type " + std::to_string(typeID));
+        }
     }
 
-    // Corrupt a valid address by replacing the 10th character with '!'.
-    // '!' is not part of the ripple alphabet.
     std::string
-    makeBadAddress(std::string good)
+    getTypeName(FieldType typeID)
```

Comment:
I would probably implement this near enum class, and move everything to anonymous namespace

Reply:
Changed in [f380124](https://github.com/XRPLF/rippled/pull/5237/commits/f380124966119c33193e8b91fdb857d521f70953)